### ISSUE TITLE
ABN loss explain

### ISSUE THREAD

**Issue by brianw0924** [realcrane/SMART#2]
Can you explain what ABN actually do? I can't understand after reading the paper and source code

elif self.attackType == 'abn':
classLoss = torch.mean((pred - flabels[batchNo*self.classifier.args.batchSize:(batchNo+1)*self.classifier.args.batchSize])**2) 

Why abn loss is not cross entropy?

**Comment by realcrane** [comment]
Hi,

As explained in the Equation 7, the ABN loss is for computing adversarial examples with its ground-truth class label falling out of the top N classes during prediction. ABN is good for attacking models that employs the top-N accuracy. Essentially, it tries to maximise the entropy of the predicted distribution over all class labels, making the predicted distribution as flat as possible. When it is perfectly flat, the predicted label is undefined because it has equal probability on every class (e.g. a softmax will give you exactly the same probability on every class label). In practice, it won't be perfect flat but still can fool the classifier.

Best,

**Comment by brianw0924** [comment]
Hi,

I understand ABN now, but why is your source code not using Entropy? For me it looks like MSE

Thank you

**Comment by realcrane** [comment]
Yes it is converted into a MSE-like loss. Given a distribution with probabilities (p_1, p_2, ..., p_n) where n is the number of classes, Shannon said the Entropy is maximized when the p_i values are equal. So the target distribution is a flat distribution with equal p_i values. This can be achieved by a MSE-like loss. Details can be found at: https://ieeexplore.ieee.org/document/6773024?reload=true&arnumber=6773024

**Comment by brianw0924** [comment]
Thanks for the explanation!

I also have a question about the other work:

[BASAR-Black-box-Attack-on-Skeletal-Action-Recognition](https://github.com/realcrane/BASAR-Black-box-Attack-on-Skeletal-
Action-Recognition)

In the Random exploration section:

Could you explain why the random directional vector is sampled from R_z*z, where z = mn/3?

I guess /3 is because three dimension(x,y,z), but why R is z*z matrix?

Thank you.

**Comment by realcrane** [comment]
Hi, if you have questions regarding another paper, can you post the question under that project? There will be others answering your questions there. I will close this question for now.
